token_reject_transaction.go

```diff
@@ -0,0 +1,349 @@
+package hedera
+
+import (
+	"time"
+
+	"github.com/hashgraph/hedera-protobufs-go/services"
+)
+
+/*-
+ *
+ * Hedera Go SDK
+ *
+ * Copyright (C) 2020 - 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+/**
+ * A transaction body to "reject" undesired tokens.<br/>
+ * This transaction will transfer one or more tokens or token
+ * balances held by the requesting account to the treasury
+ * for each token type.
+ * <p>
+ * Each transfer MUST be one of the following:
+ * <ul>
+ *   <li>A single non-fungible/unique token.</li>
+ *   <li>The full balance held for a fungible/common
+ *       token type.</li>
+ * </ul>
+ * When complete, the requesting account SHALL NOT hold the
+ * rejected tokens.<br/>
+ * Custom fees and royalties defined for the tokens rejected
+ * SHALL NOT be charged for this transaction.
+ */
+type TokenRejectTransaction struct {
+	Transaction
+	ownerID  *AccountID
+	tokenIDs []TokenID
+	nftIDs   []NftID
+}
+
+func NewTokenRejectTransaction() *TokenRejectTransaction {
+	tx := TokenRejectTransaction{
+		Transaction: _NewTransaction(),
+	}
+	return &tx
+}
+
+func _TokenRejectTransactionFromProtobuf(tx Transaction, pb *services.TransactionBody) *TokenRejectTransaction {
+	return &TokenRejectTransaction{
+		Transaction: tx,
+		ownerID:     _AccountIDFromProtobuf(pb.GetTokenReject().Owner),
+		tokenIDs:    _TokenIDsFromTokenReferenceProtobuf(pb.GetTokenReject().Rejections),
+		nftIDs:      _NftIDsFromTokenReferenceProtobuf(pb.GetTokenReject().Rejections),
```

Comment:
Don't you think that it would be better here to check the `Rejection` type(if it's ft/nft) and pass only the ft's to the `_TokenIDsFromTokenReferenceProtobuf` and the same goes for nfts and `_NftIDsFromTokenReferenceProtobuf`.

Reply:
Modified